staging/src/k8s.io/kubectl/pkg/cmd/cmd_test.go

```diff
@@ -236,19 +250,32 @@ func TestKubectlCommandHandlesPlugins(t *testing.T) {
 		t.Run(test.name, func(t *testing.T) {
 			pluginsHandler := &testPluginHandler{
 				pluginsDirectory: "plugin/testdata",
+				validPrefixes:    plugin.ValidPluginFilenamePrefixes,
 			}
 			ioStreams, _, _, _ := genericiooptions.NewTestIOStreams()
-
 			root := NewDefaultKubectlCommandWithArgs(KubectlOptions{PluginHandler: pluginsHandler, Arguments: test.args, IOStreams: ioStreams})
-			if err := root.Execute(); err != nil {
-				t.Fatalf("unexpected error: %v", err)
+			if pluginsHandler.lookedup || pluginsHandler.executed {
+				// original plugin handler is implemented by exec call so no additional actions are expected on the cobra command
+				// special case is lookedup == true && executed == false where cobra would output unknown command, but we have to check expected plugin error bellow
+			} else {
+				// args must be set, otherwise Execute will use os.Args (args used for starting the test) and test.args would not be passed
+				// to the command which might invoke only "kubectl" without any additional args and give false positives
+				root.SetArgs(test.args[1:])
+				// Important note! Incorrect command or command failing validation might just call os.Exit(1) which would interrupt execution of the test
+				if err := root.Execute(); err != nil {
+					t.Fatalf("unexpected error: %v", err)
+				}
+			}
+
+			if pluginsHandler.lookupErr != nil && pluginsHandler.lookupErr.Error() != test.expectLookupError {
```

Comment:
Where is the test case of `LookupError` is expected but not fired?

Reply:
+1, added to the condition